ingestion/src/metadata/ingestion/source/dashboard/powerbi/metadata.py

```diff
@@ -222,6 +242,103 @@ def get_chart_url(
             f"{workspace_id}/{chart_url_postfix}"
         )
 
+    def list_datamodels(self) -> Iterable[Dataset]:
+        """
+        Get All the Powerbi Datasets
+        """
+        if self.source_config.includeDataModels:
+            try:
+                for workspace in self.workspace_data:
+                    for dataset in workspace.datasets or []:
+                        if filter_by_datamodel(
+                            self.source_config.dataModelFilterPattern, dataset.name
+                        ):
+                            self.status.filter(dataset.name, "Data model filtered out.")
+                            continue
+                        yield dataset
+            except Exception as err:
+                logger.debug(traceback.format_exc())
+                logger.error(f"Unexpected error fetching PowerBI datasets - {err}")
+
+    def yield_bulk_datamodel(
+        self, dataset: Dataset
+    ) -> Iterable[CreateDashboardDataModelRequest]:
+        """
+        Method to fetch DataModels in bulk
+        """
+        try:
+            data_model_request = CreateDashboardDataModelRequest(
+                name=dataset.id,
+                displayName=dataset.name,
+                description=dataset.description,
+                service=self.context.dashboard_service.fullyQualifiedName.__root__,
+                dataModelType=DataModelType.PowerBIDataModel.value,
+                serviceType=DashboardServiceType.PowerBI.value,
+                columns=self.get_column_info(dataset),
+            )
+            yield data_model_request
+            self.status.scanned(f"Data Model Scanned: {data_model_request.displayName}")
+        except Exception as exc:
+            error_msg = f"Error yielding Data Model [{dataset.name}]: {exc}"
+            self.status.failed(
+                name=dataset.name,
+                error=error_msg,
+                stack_trace=traceback.format_exc(),
+            )
+            logger.error(error_msg)
+            logger.debug(traceback.format_exc())
+
+    def get_child_columns(self, table: PowerBiTable) -> List[Column]:
+        """
+        Extract the child columns from the fields
+        """
+        columns = []
+        for column in table.columns or []:
+            try:
+                parsed_column = {
+                    "dataTypeDisplay": column.dataType
+                    if column.dataType
+                    else DataType.UNKNOWN.value,
+                    "dataType": ColumnTypeParser.get_column_type(
+                        column.dataType if column.dataType else None
+                    ),
+                    "name": str(uuid.uuid4()),
```

Comment:
why are we generating a uuid to defined the column name?

Reply:
We are using that since a datamodel can have tables/columns with the same names. To keep the name unique we are using uuid and in the display name it is the raw name